tests/integration.rs

```diff
@@ -973,6 +973,40 @@ fn check_latest_events_exact(latest_events: &[&MetaEvent], event_file: &str) ->
     true
 }
 
+/// Return true if event occurs before timeout elapses
+///
+/// # Arguments
+///
+/// * `event` - an even to wait for
+/// * `timeout` - maximum time to wait for event
+/// * `event_file` - event file to read events from
+fn wait_for_event(event: &MetaEvent, timeout: u64, event_file: &str) -> bool {
```

Comment:
This function should be implemented by starting a new thread and having the main thread wait on a synchronisation primitive (e.g. barrier) to be released by that started thread. The started thread should use epoll on the event monitor fd to look for new messages plus a timer fd to handle a graceful timeout period.

Reply:
Upon further reflection if this solution solves the problem now (and we can demonstrate it by using it throughout the integration tests) then i'm happy to land this version and change later. i.e. lets not wait for the best version when we have something that is already better

Reply:
The current solution works for me, the only problem is that `shutdown` is already in the event file when `wait_for_event` is called, so I added extra step to check if last event [matches](https://github.com/cloud-hypervisor/cloud-hypervisor/pull/6015/commits/7d3a2bd4648ba8be66516975e6509e75ae7c3121#diff-eb771f54a1fdc1c330a82981d7d27d1202a5cbe3a59581ff67af5e3361f5f90eR987)
Then again, also my previous PR #5992 that used `check_latest_events_exact` worked for me.
I can also implement multi-threading solution as you propose. Let me know what is the best approach here.

Reply:
The way how it is done right now is simply spinning with an interval of 1s sleep. It does help with avoiding unnecessary long wait, but at the cost of creating more contention and context switch which can cause more issues to other tests with hard-coded sleeps.

I'd say it is important to find a way that does not require spinning. Also, I think we don't need to start a new thread for `wait_for_event` since the main thread need to wait anyway.

Reply:
Since shutdown event is not in events file before dbus_api.remote_command("shutdown", None) I refactored the code to run wait_for_event in a separate thread. @rbradford was right, it is needed to run wait_for_event in another thread and wait until `shutdown` is done by the main thread.